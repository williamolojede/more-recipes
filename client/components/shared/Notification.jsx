import React, { Component } from 'react';
import { CSSTransition } from 'react-transition-group';
import PropTypes from 'prop-types';

import { hideNotification } from '../../actions/notification';

class Notification extends Component {
  constructor(props) {
    super(props);
    this.state = {
      timeout: 3000
    };
  }
  componentWillReceiveProps(newprops) {
    if (newprops.notification) {
      setTimeout(() => {
        this.props.dispatch(hideNotification());
      }, this.state.timeout);
    }
  }

  render() {
    const { notification } = this.props;
    return (
      <CSSTransition
        classNames="notification"
        in={!!notification}
        timeout={this.state.timeout}
      >
        <div className="notification z-depth-1">
          <p>{notification}</p>
        </div>
      </CSSTransition>
    );
  }
}

Notification.propTypes = {
  notification: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired
};
export default Notification;
