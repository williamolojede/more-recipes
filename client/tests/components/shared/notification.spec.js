import React from 'react';
import renderer from 'react-test-renderer';
import Notification from '../../../components/shared/Notification';

function setup() {
  const props = {
    notification: 'you are testing this',
    dispatch: jest.fn()
  };
  const component = renderer.create(<Notification {...props} />);
  return {
    component
  };
}


describe('Notification', () => {
  it('should render with right amount of elements', () => {
    const tree = setup().component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
