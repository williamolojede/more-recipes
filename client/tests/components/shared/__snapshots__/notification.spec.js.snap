// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render with right amount of elements 1`] = `
<CSSTransition
  classNames="notification"
  in={true}
  timeout={3000}
>
  <div
    className="notification z-depth-1"
  >
    <p>
      you are testing this
    </p>
  </div>
</CSSTransition>
`;

exports[`Notification should render with right amount of elements when notifiacation is empty 1`] = `
<CSSTransition
  classNames="notification"
  in={false}
  timeout={3000}
>
  <div
    className="notification z-depth-1"
  >
    <p />
  </div>
</CSSTransition>
`;
